buildenv/buildenv.nix

```diff
@@ -0,0 +1,278 @@
+{
+  activationScripts_storePath ? "@activationScripts@",
+  defaultEnvrc_storePath ? "@defaultEnvrc@",
+  coreutils_storePath ? "@coreutils@",
+  floxBuildenv_storePath ? "@out@",
+  manifest,
+  name ? "environment",
+  serviceConfigYaml ? "",
+}:
+let
+
+  # Start by converting (text) storePaths into contextual strings.
+  # FTR a contextual string is one that knows it's a derivation!
+  # This is extremely important for populating the Nix dependency
+  # graph. See `builtins.getContext` and `addContextFrom` in nixpkgs
+  # strings.nix.
+  activationScripts = builtins.storePath activationScripts_storePath;
+  defaultEnvrc = builtins.storePath defaultEnvrc_storePath;
+  floxBuildEnv = builtins.storePath floxBuildenv_storePath;
+  coreutils = builtins.storePath coreutils_storePath;
+
+  # A helpful library function copied from nixpkgs/lib/attrsets.nix.
+  # foldlAttrs = f: init: set:
+  #   builtins.foldl'
+  #     (acc: name: f acc name set.${name})
+  #     init
+  #     (builtins.attrNames set);
+
+  # The system we're building for.
+  system = builtins.currentSystem;
+
+  # Copy manifest file into the store for access within derivations.
+  manifestLock = /. + manifest;
+
+  # Parse the manifest file.
+  manifestLockData = builtins.fromJSON (builtins.readFile manifest);
+  manifestData = manifestLockData.manifest;
+
+  build = if (builtins.hasAttr "build" manifestData) then manifestData.build else { };
+  hook = if (builtins.hasAttr "hook" manifestData) then manifestData.hook else { };
+  profile = if (builtins.hasAttr "profile" manifestData) then manifestData.profile else { };
+  vars =
+    if (builtins.hasAttr "vars" manifestData) then
+      (builtins.toFile "envrc-vars" (
+        builtins.concatStringsSep "" (
+          builtins.map (n: "export ${n}=\"${builtins.getAttr n manifestData.vars}\"\n") (
+            builtins.attrNames manifestData.vars
+          )
+        )
+        # alternative ... worth it?
+        #      foldlAttrs (
+        #        acc: n: v: acc + "export ${n}=\"${v}\"\n"
+        #      ) "" manifestData.vars
+      ))
+    else
+      null;
+
+  # Calculate environment outputs.
+  environmentOutputs = [
+    "runtime"
+    "develop"
+  ] ++ (builtins.map (x: "build-${x}") (builtins.attrNames build));
+
+  createManifestChunks =
+    [
+      # static chunks
+      ''
+        export PATH="${coreutils}/bin''${PATH:+:}''${PATH}"
+        mkdir -p $out/activate.d
+        cp --no-preserve=mode ${manifestLock} $out/manifest.lock
+        cp --no-preserve=mode ${defaultEnvrc} $out/activate.d/envrc
```

Comment:
should we `ln -s` these instead? if only to quickly diff environments with nix for dev purposes?

Reply:
As above, would prefer not. That and the manifest.lock will usually be changing with each iteration because that's the most often trigger for rebuilding in the first place, and we can't link the envrc as we need to append to it below.

Reply:
what's the envrc anyway?
ist that the new name for the `activate` file?

Reply:
`$FLOX_ENV/activate/envrc` is the file that contains all of the a) default env variable assignments and b) static env var assignments set in the `[vars]` section of the manifest. It is sourced by the `activate` script.